include/RAJA/policy/hip/kernel/For.hpp

```diff
@@ -686,35 +545,36 @@ struct HipStatementExecutor<
   using diff_t = segment_diff_type<ArgumentId, Data>;
 
 
-  static
-  inline RAJA_DEVICE void exec(Data &data, bool thread_active)
+  static inline RAJA_DEVICE
+  void exec(Data &data, bool thread_active)
   {
     // grid stride loop
     diff_t len = segment_length<ArgumentId>(data);
-    diff_t i_init = get_hip_dim<BlockDim>(dim3(blockIdx.x,blockIdx.y,blockIdx.z));
-    diff_t i_stride = get_hip_dim<BlockDim>(dim3(gridDim.x,gridDim.y,gridDim.z));
+    diff_t i_init = Indexer::template index<diff_t>();
+    diff_t i_stride = Indexer::template size<diff_t>();
 
-    // Iterate through grid stride of chunks
-    for(diff_t i = i_init;i < len;i += i_stride){
+    // Iterate through chunks
+    for (diff_t ii = 0; ii < len; ii += i_stride) {
+      diff_t i = ii + i_init;
```

Comment:
This will cause more instructions to be issued for block Tiles. This pattern comes from thread tiles to avoid warp divergence.